File: andre/cpp/uva10920.cpp
/**
 * Working on:
 * 24122150	10920	Spiral Tap	Accepted	C++11	0.420	2019-10-29 15:44:28
 **/

#include <iostream>
#include <math.h> 

using namespace std;

enum Side
{
	Up = 1,
	Left = 2,
	Down = 3,
	Right = 4
};

int main()
{
	unsigned int boardSize, middleLine, middleColumn;
	unsigned long long int  desiredPosition, stepSize, stepsDone;
	bool incrementStep;
	int currentSide, currentPos;
	int sides[4] = { Side::Up, Side::Left, Side::Down, Side::Right };

	cin >> boardSize >> desiredPosition;

	while (boardSize > 0 && desiredPosition > 0) {
		middleLine = middleColumn = ceil(boardSize / 2.0);

		stepSize = stepsDone = 1;
		incrementStep = false;
		currentPos = 0;

		while (stepsDone < desiredPosition) {
			currentSide = sides[currentPos];
			if (++currentPos == 4) currentPos = 0;

			switch (currentSide)
			{
			case Side::Up:
				middleLine = middleLine + stepSize;
				break;

			case Side::Left:
				middleColumn = middleColumn - stepSize;
				incrementStep = true;
				break;

			case Side::Down:
				middleLine = middleLine - stepSize;
				break;

			case Side::Right:
				middleColumn = middleColumn + stepSize;
				incrementStep = true;
				break;
			}

			stepsDone += stepSize;

			if (incrementStep) {
				stepSize = stepSize + 1;
				incrementStep = false;
			}
		}

		if (stepsDone > desiredPosition) {
			int stepBack = stepsDone - desiredPosition;

			switch (currentSide)
			{
			case Side::Up:
				middleLine = middleLine - stepBack;
				break;

			case Side::Left:
				middleColumn = middleColumn + stepBack;
				break;

			case Side::Down:
				middleLine = middleLine + stepBack;
				break;

			case Side::Right:
				middleColumn = middleColumn - stepBack;
				break;
			}
		}

		cout << "Line = " << middleLine << ", column = " << middleColumn << ".\n";
		cin >> boardSize >> desiredPosition;
	}
}

File: andre/py/uva10920.py
# uva10920
from itertools import cycle
from math import ceil
from enum import Enum

class Side(Enum):
    Up = 1
    Left = 2
    Down = 3
    Right = 4
#end-class

def main():
    inputData = input().split(' ')

    boardSize = int(inputData[0])
    position = int(inputData[1])
    
    while(boardSize != 0 and position != 0):
        side = cycle([Side.Up, Side.Left, Side.Down, Side.Right])
        middle = ceil(boardSize/2.0)
        middleLine = middleColumn = middle

        stepSize = 1
        stepsDone = 1
        incrementStep = False
        actualSide = Side.Up #Just to declare Var
        while(stepsDone < position):
            actualSide = next(side)

            if(actualSide == Side.Up):
                middleLine = middleLine + stepSize
            #end-if
            elif(actualSide == Side.Left):
                middleColumn = middleColumn - stepSize
                incrementStep = True
            #end-elif
            elif(actualSide == Side.Down):
                middleLine = middleLine - stepSize
            #end-elif
            else:
                middleColumn = middleColumn + stepSize
                incrementStep = True
            #end-if

            stepsDone = stepsDone + stepSize

            if(incrementStep):
                stepSize = stepSize + 1
                incrementStep = False
            #end-if
        #end-while

        if(stepsDone > position):
            stepBack = stepsDone - position

            if(actualSide == Side.Up):
                middleLine = middleLine - stepBack
            #end-if
            elif(actualSide == Side.Left):
                middleColumn = middleColumn + stepBack
            #end-elif
            elif(actualSide == Side.Down):
                middleLine = middleLine + stepBack
            #end-elif
            else:
                middleColumn = middleColumn - stepBack
            #end-if
        #end-if

        print('Line = %d, column = %d.' % (middleLine, middleColumn))

        inputData = input().split(' ')
        boardSize = int(inputData[0])
        position = int(inputData[1])
    #end-while
# end-main()


if __name__ == "__main__":
    main()
# end-if
